### PR TITLE
fix clang "parameter not found" warnings

### PR DESCRIPTION
**Pull request by samuela**
Pretty simple.

### CODE REVIEW

File: include/boost/algorithm/string/formatter.hpp [hunk 1]
```diff
@@ -83,7 +83,7 @@ namespace boost {
         template<typename RangeT>
         inline detail::empty_formatF< 
             BOOST_STRING_TYPENAME range_value<RangeT>::type>
-        empty_formatter(const RangeT&)
+        empty_formatter(const RangeT& Input)
```
Comment (mclow) [inline include/boost/algorithm/string/formatter.hpp]: I don't think the parameter is used; so naming it here would probably generate other warnings; something like "input not used"

Comment (samuela) [reply include/boost/algorithm/string/formatter.hpp]: Mmm, true. I didn't observe such a warning but I'm sure they would come up. The docstring "parameter not found" doesn't like unnamed arguments as far as I can tell.

Which warning is preferred? Should I revert this one?

Comment (mclow) [reply include/boost/algorithm/string/formatter.hpp]: I'd rather have a docstring warning than a compiler warning.

Comment (mclow) [reply include/boost/algorithm/string/formatter.hpp]: The other way to deal with it is to add `(void) Input;` to the body of the function.

Comment (samuela) [reply include/boost/algorithm/string/formatter.hpp]: Ok, reverted in c34713b!

Comment (mclow) [reply include/boost/algorithm/string/formatter.hpp]: thanks!

**Pull request merged**